refactor(forms): remove ngForm element selector

crisbeto opened the pull request:
Removes the deprecated `ngForm` element selector and all of the code related to it.

BREAKING CHANGES:
* `<ngForm></ngForm>` can no longer be used as a selector. Use `<ng-form></ng-form>` instead.
* The `NgFromSelectorWarning` directive has been removed.
* `FormsModule.withConfig` has been removed. Use the `FormsModule` directly.

crisbeto commented on packages/forms/src/form_providers.ts:
```diff
@@ -31,18 +31,8 @@ export class FormsModule {
-          [{provide: NG_FORM_SELECTOR_WARNING, useValue: opts.warnOnDeprecatedNgFormSelector}]
-    };
-  }
+  static withConfig(opts: {}): ModuleWithProviders<FormsModule> { return {ngModule: FormsModule}; }
```
note for reviewer: should I get rid of this method entirely?

kara replied:
Yes, let's remove it entirely

kara commented on aio/content/guide/deprecations.md:
```diff
@@ -408,6 +403,8 @@ The following APIs have been removed starting with version 8.0.0:
 | `@angular/core` | [`Renderer`](https://v8.angular.io/api/core/Renderer) | [`Renderer2`](https://angular.io/api/core/Renderer2) | [Migration guide.](guide/migration-renderer)
 | `@angular/core` | [`RootRenderer`](https://v8.angular.io/api/core/RootRenderer) | [`RendererFactory2`](https://angular.io/api/core/RendererFactory2) | none
 | `@angular/core` | [`RenderComponentType`](https://v8.angular.io/api/core/RenderComponentType) | [`RendererType2`](https://angular.io/api/core/RendererType2) | none
+| `@angular/forms` | [`NgFormSelectorWarning`](https://v8.angular.io/api/forms/NgFormSelectorWarning) | none | none
```
Can you start a separate section for v9? The renderer tokens should probably be in the v9 section too.

```suggestion
The following APIs have been removed starting with version 9.0.0:

| `@angular/forms` | [`NgFormSelectorWarning`](https://v8.angular.io/api/forms/NgFormSelectorWarning) | none | none
```

crisbeto commented:
The feedback has been addressed.

AndrewKushnir commented:
- [VE Presubmit](http://test/OCL:273878081:BASE:273878092:1570677669833:b60e9988)
- [Global VE Presubmit](https://test.corp.google.com/OCL:273878081:BASE:273878172:1570677722153:9ab58c15)

AndrewKushnir commented:
FYI, VE presubmits (including a [global run](https://test.corp.google.com/ui#id=OCL:273878081:BASE:273970351:1570723912903:68373d01)) are successful - no new regressions. Thank you.

kara reviewed (approved):
LGTM

kara commented:
merge assistance: global approval

kara commented on aio/content/guide/deprecations.md:
```diff
@@ -405,9 +400,17 @@ The following APIs have been removed starting with version 8.0.0:
+| `@angular/forms` | `ngForm` element selector | `ng-form` element selector | none
 
 
 <!-- The following anchor is used by redirects from the removed API pages. Do not change or remove. -->
```
I just remembered that we have a section for this `ngForm` warning here:

https://github.com/angular/angular/blob/5efa91f4a44489ea9bb0fe63a0b5a12f4b8653c3/aio/content/guide/deprecations.md#ngform-element-selector

Can we remove that?

crisbeto replied:
Done. Initially I just removed the references to the other doc pages.

kara reviewed (changes requested):
Sorry, one more thing

crisbeto committed:
refactor(forms): remove ngForm element selector

Removes the deprecated `ngForm` element selector and all of the code related to it.

BREAKING CHANGES:
* `<ngForm></ngForm>` can no longer be used as a selector. Use `<ng-form></ng-form>` instead.
* The `NgFromSelectorWarning` directive has been removed.
* `FormsModule.withConfig` has been removed. Use the `FormsModule` directly.

kara reviewed (approved):
LGTM

kara commented:
merge-assistance: global approval
